src/AlertBox/AlertBox.tsx

```diff
@@ -24,10 +33,25 @@ const cssClass = {
   CLOSE: "AlertBox--close",
 };
 
+const TYPES = {
+  WARNING: "warning",
+  SUCCESS: "success",
+  ERROR: "error",
+  INFO: "info",
+};
+
 /*
  * AlertBox is a closable, highlighted box
  */
-export default class AlertBox extends React.PureComponent {
+export default class AlertBox extends React.PureComponent<Props> {
+  static propTypes = {
```

Comment:
Now that we have real TypeScript types, is there any reason to keep old React prop types around?

Reply:
Runtime type checking still provides some value, though the value is small since we often ignore the console warnings 🙈. I don't know if they're worth it for type checking. I'm mainly hesitant to remove them because I tried to avoid too many code changes in this pull request (yeah I know I still ended up with a lot of diffs). There are a bunch of places where we filter props based on `propTypes` like https://github.com/Clever/components/blob/031822e606f15f4d9e07efc890c8e70657e3d702/src/TopBar/TopBarButton.tsx#L31, so removing them wouldn't be trivial. I wouldn't be opposed to removing them in the future.

Reply:
Yeah my gut feeling is that we don't need React `propTypes` moving forward. The overhead of keeping two type definitions in sync doesn't feel worth the marginal benefit of runtime checking that we typically ignore :smile:

But great point that removing `propTypes` is a non-trivial change. We can leaving existing components as is.

Let's just update the new component script to not include `propTypes`

Reply:
fwiw I discovered through local testing that changing
```
<SubComponent
  {...additionalProps} {/* props - propTypes */}
  ...
/>
```
to
```
<SubComponent
  {...this.props}
  ...
/>
```
is a breaking change

Reply:
ugh I'm afraid to change any of that props filtering code